felix/dataplane/linux/bpf_ep_mgr.go

```diff
@@ -561,17 +679,48 @@ func (m *bpfEndpointManager) updateIfaceStateMap(name string, iface *bpfInterfac
 		if iface.dpState.isReady {
 			flags |= ifstate.FlgReady
 		}
-		v := ifstate.NewValue(flags, name)
+		v := ifstate.NewValue(flags, name,
+			iface.dpState.policyIdx[hook.XDP],
+			iface.dpState.policyIdx[hook.Ingress],
+			iface.dpState.policyIdx[hook.Egress])
 		m.ifStateMap.SetDesired(k, v)
 	} else {
+		if err := m.policyMapDelete(hook.XDP, iface.dpState.policyIdx[hook.XDP]); err != nil {
```

Comment:
not sure if I see what you mean, but I removed the explicity `iface.dpState.policyIdx[*] = -1` as it is duplicated by the clearPolicies at the end.

`policyMapAlloc.Get()` is called on a different level than `policyMapAlloc.Put()` because of how the original code was structured and the differences between HEPs and WEPs. We can have another pass on this later.